App/View Controllers/Posts/PostsPageViewController.swift

```diff
@@ -33,6 +33,51 @@ final class PostsPageViewController: ViewController {
     private var scrollToFractionAfterLoading: CGFloat?
     let thread: AwfulThread
     private var webViewDidLoadOnce = false
+    
+    lazy var threadActionsMenu: UIMenu = {
+        var threadActions: [UIMenuElement] = []
+        
+        let threadActionMenu: UIMenu = {
+            let bookmarkTitle = self.thread.bookmarked ? "Remove Bookmark" : "Bookmark Thread"
+            let bookmarkImage = self.thread.bookmarked ?
+            UIImage(named: "remove-bookmark")!.withRenderingMode(.alwaysTemplate)
+            :
+            UIImage(named: "add-bookmark")!.withRenderingMode(.alwaysTemplate)
+            let yourPostsImage = UIImage(named: "single-users-posts")!.withRenderingMode(.alwaysTemplate)
+            let copyURLImage = UIImage(named: "copy-url")!.withRenderingMode(.alwaysTemplate)
+            let voteImage = UIImage(named: "vote")!.withRenderingMode(.alwaysTemplate)
+            
+            // Copy link
+            let copyLink = UIAction.Identifier("copyLink")
+            actionMappings[copyLink] = copyLink(action:)
+            let copyLinkAction = UIAction(title: "Copy link", image: copyURLImage, identifier: copyLink, handler: copyLink(action:))
+            threadActions.append(copyLinkAction)
+            
+            // Vote
+            let vote = UIAction.Identifier("vote")
+            actionMappings[vote] = vote(action:)
+            let voteAction = UIAction(title: "Vote", image: voteImage, identifier: vote, handler: vote(action:))
+            threadActions.append(voteAction)
+            
+            // Your posts
+            let yourPosts = UIAction.Identifier("yourPosts")
+            actionMappings[yourPosts] = yourPosts(action:)
+            let yourPostsAction = UIAction(title: "Your posts", image: yourPostsImage, identifier: yourPosts, handler: yourPosts(action:))
+            threadActions.append(yourPostsAction)
+            
+            // Remove bookmark
+            let bookmark = UIAction.Identifier("bookmark")
+            actionMappings[bookmark] = bookmark(action:)
+            let bookmarkAction = UIAction(title: bookmarkTitle, image: bookmarkImage, identifier: bookmark, handler: bookmark(action:))
+            bookmarkAction.attributes = self.thread.bookmarked ? [.destructive] : []
+            threadActions.append(bookmarkAction)
+            
+            let tempMenu = UIMenu(title: "", image: nil, identifier: nil, options: [.displayInline], children: threadActions)
+            return UIMenu(title: self.thread.title ?? "", image: nil, identifier: nil, options: [.displayInline], children: [tempMenu])
+        }()
+        return threadActionMenu
+    }()
+    
```

Comment:
I was thinking this menu looked a little janky. Appreciate making it proper UIMenu :)

Reply:
:)